public/docs/ts/_cache/tutorial/toh-pt5.jade

```diff
@@ -359,49 +429,61 @@ code-example(language="bash").
 
   Although we display the details of a selected hero at the bottom of the `HeroesComponent`,
   we don't yet *navigate* to the `HeroDetailComponent` in the three ways specified in our requirements:
+
   1. from the *Dashboard* to a selected hero.
   1. from the *Heroes* list to a selected hero.
   1. from a "deep link" URL pasted into the browser address bar.
 
-  Adding a `'HeroDetail'` route seem an obvious place to start.
+  Adding a hero-detail route seems like an obvious place to start.
 
   ### Routing to a hero detail
 
-  We'll add a route to the `HeroDetailComponent` in the `app.routing.ts` where our other routes are configured.
+  We'll add a route to the `HeroDetailComponent` in `!{_appRoutingTsVsAppComp}` where our other routes are configured.
 
   The new route is a bit unusual in that we must tell the `HeroDetailComponent` *which hero to show*.
   We didn't have to tell the `HeroesComponent` or the `DashboardComponent` anything.
 
-  At the moment the parent `HeroesComponent` sets the component's `hero` property to a hero object with a binding like this.
-code-example(format='').
+  At the moment the parent `HeroesComponent` sets the component's `hero` property to a 
+  hero object with a binding like this.
+
+code-example(language="html").
   &lt;my-hero-detail [hero]="selectedHero">&lt;/my-hero-detail>
+
 :marked
   That clearly won't work in any of our routing scenarios.
   Certainly not the last one; we can't embed an entire hero object in the URL! Nor would we want to.
 
   ### Parameterized route
-  We *can* add the hero's `id` to the URL. When routing to the hero whose `id` is 11, we could expect to see an URL such as this:
+
+  We *can* add the hero's `id` to the URL. When routing to the hero whose `id` is 11, 
+  we could expect to see an URL such as this:
+
 code-example(format='').
   /detail/11
+
 :marked
   The `/detail/` part of that URL is constant. The trailing numeric `id` part changes from hero to hero.
   We need to represent that variable part of the route with a *parameter* (or *token*) that stands for the hero's `id`.
 
   ### Configure a Route with a Parameter
 
   Here's the *route definition* we'll use.
-+makeExample('toh-5/ts/app/app.routing.2.ts','hero-detail-route', 'app/app.routing.ts (route to HeroDetailComponent)')(format=".")
+
+- var _file = _docsFor == 'dart' ? 'app/app.component.ts' : 'app/app.routing.ts'
++makeExcerpt(_file + ' (hero detail)','hero-detail')
+
 :marked
   The colon (:) in the path indicates that `:id` is a placeholder to be filled with a specific hero `id`
   when navigating to the `HeroDetailComponent`.
 .l-sub-section
   :marked
-    Of course we have to import the `HeroDetailComponent` before we create this route:
-  +makeExample('toh-5/ts/app/app.routing.2.ts','hero-detail-import')(format=".")
-:marked
-  Add the `HeroDetailComponent` to our root NgModule's `declarations`.
+    Remember to import `HeroDetailComponent` before creating this route.
```

Comment:
You're not really importing that class, you're importing the file that contains it. Why not show the import, either here or in the "hero detail" code above?